File: RedSheet.DbEntities/Models/vProjectCulturePlanRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectCulturePlanRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectCulturePlanRecord  
	{
 
	[Range(1, int.MaxValue)]
	public int CulturePlanCategoryId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectCulturePlanId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string ProjectCulturePlanValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 

	public vProjectCulturePlanRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-project-culture-plan-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectCulturePlanRecord {
    constructor(vProjectCulturePlanRecord?: vProjectCulturePlanRecord )  {
        let properties = [ "culturePlanCategoryId", "projectCulturePlanId", "projectCulturePlanValue", "projectModuleId",];
        for (let property of properties)
            if (vProjectCulturePlanRecord && vProjectCulturePlanRecord[property])
                this[property] = vProjectCulturePlanRecord[property];
    }
 
    @range(1,2147483647)
	culturePlanCategoryId : number =   undefined;
 
	projectCulturePlanId : number =   0 ;
 
    @required()
    @maxLength(200)
	projectCulturePlanValue : string =   undefined;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;


}
